File: problems/2644_find_the_maximum_divisibility_score/solution.py
from typing import List


class Solution:
    def maxDivScore(self, nums: List[int], divisors: List[int]) -> int:
        best_d, best_cnt = divisors[0], -1
        for d in divisors:
            cnt = sum(1 for n in nums if n % d == 0)
            if cnt > best_cnt or (cnt == best_cnt and d < best_d):
                best_d, best_cnt = d, cnt
        return best_d

File: problems/2644_find_the_maximum_divisibility_score/Solution.java
class Solution {
    public int maxDivScore(int[] nums, int[] divisors) {
        int bestD = divisors[0], bestCnt = -1;
        for (int d : divisors) {
            int cnt = 0; for (int n : nums) if (n % d == 0) cnt++;
            if (cnt > bestCnt || (cnt == bestCnt && d < bestD)) { bestD = d; bestCnt = cnt; }
        }
        return bestD;
    }
}
